diff --git a/src/rpc/mining.cpp b/src/rpc/mining.cpp
index 8625062..07c4bda 100644
--- a/src/rpc/mining.cpp
+++ b/src/rpc/mining.cpp
@@ -21,6 +21,9 @@
 #include "util.h"
 #include "utilstrencodings.h"
 #include "validationinterface.h"
+#include "oerushield/oerudb.h"
+#include "oerushield/oerushield.h"
+#include "oerushield/oerusignal.h"

 #include <stdint.h>

@@ -689,6 +692,11 @@ UniValue getblocktemplate(const UniValue& params, bool fHelp)
         result.push_back(Pair("default_witness_commitment", HexStr(pblocktemplate->vchCoinbaseCommitment.begin(), pblocktemplate->vchCoinbaseCommitment.end())));
     }

+    COeruShield oeruShield(poeruDBMain);
+    int blocksSinceLastCertified = oeruShield.GetBlocksSinceLastCertified(pblock, pindexPrev) + 1;
+    bool mining_disabled = !(blocksSinceLastCertified >= 0 && blocksSinceLastCertified < Params().OeruShieldMaxBlocksSinceLastCertified());
+    result.push_back(Pair("mining_disabled", (mining_disabled)? true : false ));
+
     return result;
 }
