diff --git a/src/rpc/mining.cpp b/src/rpc/mining.cpp
index 3b580a5a3..1ae960b13 100644
--- a/src/rpc/mining.cpp
+++ b/src/rpc/mining.cpp
@@ -25,6 +25,7 @@
 #include <utilstrencodings.h>
 #include <validationinterface.h>
 #include <warnings.h>
+#include <wallet/wallet.h>
 
 #include <memory>
 #include <stdint.h>
@@ -773,13 +774,15 @@ UniValue submitblock(const JSONRPCRequest& request)
 CKeyID GetAuxpowMiningKey()
 {
     CKeyID result;
-    // CBitcoinSecret auxminingaddr(gArgs.GetArg("-auxminingaddr", ""));
-    // if (!auxminingaddr.GetKeyID(result)) {
-    //     CReserveKey reservekey(vpwallets[0]);
-    //     CPubKey pubkey;
-    //     reservekey.GetReservedKey(pubkey);
-    //     result = pubkey.GetID();
-    // }
+
+//    CBitcoinSecret auxminingaddr(gArgs.GetArg("-auxminingaddr", ""));
+//    if (!auxminingaddr.GetKeyID(result)) {
+    	CReserveKey reservekey(vpwallets[0]);
+        CPubKey pubkey;
+        reservekey.GetReservedKey(pubkey);
+        result = pubkey.GetID();
+//    }
+
     return result;
 }
 
