diff --git a/src/bitcoinrpc.cpp b/src/bitcoinrpc.cpp
index 1f4d00c..23cf206 100644
--- a/src/bitcoinrpc.cpp
+++ b/src/bitcoinrpc.cpp
@@ -2260,6 +2260,37 @@ Value getblocktemplate(const Array& params, bool fHelp)
         result.push_back(Pair("bits", HexBits(pblock->nBits)));
         result.push_back(Pair("height", (int64_t)(pindexPrev->nHeight+1)));
 
+        bool mining_disabled = false;
+        uint nOurTime   = (uint)GetAdjustedTime();
+        int nHeight = pindexPrev->nHeight+1;
+        int64 nTime = pblock->nTime;
+
+        if(nHeight >= nSoftForkOne) {
+
+           if(nTime > (nOurTime + 10 * 60))
+        	   mining_disabled = true;
+
+           if(nTime <= (pindexPrev->GetMedianTimePast() + BLOCK_LIMITER_TIME))
+        	   mining_disabled = true;
+
+           if(nTime <= (pindexPrev->GetBlockTime() - 10 * 60))
+        	   mining_disabled = true;
+        }
+        /* Soft fork 2 */
+        if(nHeight >= nSoftForkTwo) {
+
+        	/* Check for time stamp (future limit) */
+            if(nTime > (nOurTime + 5 * 60))
+         	   mining_disabled = true;
+
+            /* Future travel detector for the block limiter */
+            if((nTime > (nOurTime + 60)) && ((pindexPrev->GetAverageTimePast(5, 45) + BLOCK_LIMITER_TIME) > nOurTime))
+         	   mining_disabled = true;
+
+        }
+
+        result.push_back(Pair("mining_disabled", (mining_disabled)? true : false ));
+
         return result;
     }
     else
diff --git a/src/makefile.unix b/src/makefile.unix
index 187680d..d2a63da 100644
--- a/src/makefile.unix
+++ b/src/makefile.unix
@@ -2,7 +2,7 @@
 # Distributed under the MIT/X11 software license, see the accompanying
 # file license.txt or http://www.opensource.org/licenses/mit-license.php.
 
-USE_UPNP:=0
+USE_UPNP:=-
 
 DEFS=-DUSE_IPV6 -DBOOST_SPIRIT_THREADSAFE
 
