diff --git a/src/chainparams.cpp b/src/chainparams.cpp
index 9815ca0..56860e1 100644
--- a/src/chainparams.cpp
+++ b/src/chainparams.cpp
@@ -277,9 +277,14 @@ public:
             ( 1154619, uint256S("0xb3df98957ccfeb9da0eb5ef929cf929ae9a6bf1917f79578688b55c25cf6b67d"))
             ( 1213390, uint256S("0x098ed783d5cf8428215686ddef1a88ba372d58b28f3f571a437f15bbaae59d55"))     
             ( 1216820, uint256S("0xe69b85a94471d42b254be77062f65edba84f3a5972a1d2a4bb8b14e1526cef52"))			
-            ( 1224490, uint256S("c20d8dab92dc16854d6cddefe9ce7905aa18b0ad949cc7848dc7008338641c4c")),
-            1513895220, // * UNIX timestamp of last checkpoint block
-            1283706,   // * total number of transactions between genesis and last checkpoint
+            ( 1224490, uint256S("0xc20d8dab92dc16854d6cddefe9ce7905aa18b0ad949cc7848dc7008338641c4c"))
+            ( 2192860, uint256S("0x0916d22a70d1b934a31934a2ea1482604c594bf3a1a9e58a73a627fde91dfebd"))
+            ( 2210000, uint256S("0x4f58bae8c6f572e7821bd2d3e6c66f2f82dce048b31302ef44f8ca8fd9ac139f"))
+            ( 2260000, uint256S("0xd04a48d6324eaf14e86c9de975eccf32449d6d6a3beb595868fdd273af1d9502"))
+            ( 2300000, uint256S("0x98bdbd3a6fa9d9daa714b3197506e905f0802866ef1bd8c669aba20ec19dfc1c"))
+			( 2358115, uint256S("0x2eb558ba005a7f3ff2cbcb2ce3a1dc1384c824f9ac401aa4c3fa1c9ebe19f44e")),
+			1591084557, // * UNIX timestamp of last checkpoint block
+			2475276,   // * total number of transactions between genesis and last checkpoint
                         //   (the tx=... number in the SetBestChain debug.log lines)
             1440.0     // * estimated number of transactions per day after checkpoint
         };
