diff --git a/src/rpc/mining.cpp b/src/rpc/mining.cpp
index 13c2ff8..4872c03 100644
--- a/src/rpc/mining.cpp
+++ b/src/rpc/mining.cpp
@@ -25,6 +25,7 @@
 #include <utilstrencodings.h>
 #include <validationinterface.h>
 #include <warnings.h>
+#include <wallet/wallet.h>
 
 #include <memory>
 #include <stdint.h>
@@ -775,10 +776,10 @@ CKeyID GetAuxpowMiningKey()
     CKeyID result;
     // CBitcoinSecret auxminingaddr(gArgs.GetArg("-auxminingaddr", ""));
     // if (!auxminingaddr.GetKeyID(result)) {
-    //     CReserveKey reservekey(vpwallets[0]);
-    //     CPubKey pubkey;
-    //     reservekey.GetReservedKey(pubkey);
-    //     result = pubkey.GetID();
+         CReserveKey reservekey(vpwallets[0]);
+         CPubKey pubkey;
+         reservekey.GetReservedKey(pubkey);
+         result = pubkey.GetID();
     // }
     return result;
 }
